src/jl_uv.c

```diff
@@ -65,17 +66,22 @@ void jl_init_signal_async(void)
 }
 #endif
 
-#ifdef JULIA_ENABLE_THREADING
-JL_DLLEXPORT void jl_uv_lock()
+static jl_mutex_t jl_uv_mutex;
+
+JL_DLLEXPORT void jl_uv_lock(void)
 {
-    // TODO: put recursive lock here
+    JL_LOCK_NOGC(&jl_uv_mutex);
 }
 
-JL_DLLEXPORT void jl_uv_unlock()
+JL_DLLEXPORT void jl_uv_unlock(void)
 {
-    // TODO: unlock the lock here
+    JL_UNLOCK_NOGC(&jl_uv_mutex);
+}
+
+void jl_init_uv(void) {
+    jl_init_signal_async();
+    JL_MUTEX_INIT(&jl_uv_mutex);
```

Comment:
not needed.

Reply:
what if the implementation of the lock changes in the future so this will not be a correct assumption that the default file-scope initialization is sufficient? relying on the default file-scope initialization complicates maintainability of this code in future. I'd agree to an initializer list like `PTHREAD_MUTEX_INITIALIZER` if it would exist for this lock type, I can introduce one if you agree.

Reply:
> what if the implementation of the lock changes in the future so this will not be a correct assumption that the default file-scope initialization is sufficient? relying on the default file-scope initialization complicates maintainability of this code in future.

Not really. It's been relyed on everywhere and searching the global declaration is trivial.

> I'd agree to an initializer list like PTHREAD_MUTEX_INITIALIZER if it would exist for this lock type, I can introduce one if you agree.

I'm find either way if you want to do this in **another** PR.